tmt/steps/report/reportportal.py

```diff
@@ -41,13 +43,51 @@ def _str_env_to_default(option: str, default: Optional[str]) -> Optional[str]:
     return str(os.getenv(env_var))
 
 
-def _filter_invalid_chars(data: str) -> str:
+@dataclasses.dataclass
+class LogFilterSettings:
+    size: int = DEFAULT_LOG_SIZE_LIMIT
+    is_traceback: bool = False
+
+
+def _filter_invalid_chars(data: str,
+                          settings: LogFilterSettings) -> str:
     return re.sub(
         '[^\u0020-\uD7FF\u0009\u000A\u000D\uE000-\uFFFD\U00010000-\U0010FFFF]+',
         '',
         data)
 
 
+def _filter_log_per_size(data: str,
+                         settings: LogFilterSettings) -> str:
+    size = len(data)
+    if size > settings.size:
+        if settings.is_traceback:
+            variable = "TMT_PLUGIN_REPORT_REPORTPORTAL_TRACEBACK_SIZE_LIMIT"
+            option = "--traceback-size-limit"
+        else:
+            variable = "TMT_PLUGIN_REPORT_REPORTPORTAL_LOG_SIZE_LIMIT"
+            option = "--log-size-limit"
+        header = (f"WARNING: Uploaded log has been truncated because its size {size} bytes "
+                  f"exceeds tmt reportportal plugin limit of {settings.size} bytes."
+                  f"The limit is controlled with {option} plugin option or"
```

Comment:
If I am not mistaken, there will be no space between these sentences. Shouldn't it be like this?
```suggestion
                  f"exceeds tmt reportportal plugin limit of {settings.size} bytes. "
                  f"The limit is controlled with {option} plugin option or "
```